comp/core/flare/helpers/send_flare.go

```diff
@@ -34,7 +34,13 @@ type flareResponse struct {
 	Error  string `json:"error,omitempty"`
 }
 
-func getFlareReader(multipartBoundary, archivePath, caseID, email, hostname, source string) io.ReadCloser {
+// FlareSource has metadata about why the flare was sent
+type FlareSource struct {
+	SourceType string
+	RCTaskUUID string
+}
```

Comment:
As there are only two kinds of `FlareSource`, I suggest having two constructors:  `NewLocalFlareSource()` and `NewRemoteConfigFlareSource(rcTaskUUID string)`

Note: The field of `FlareSource` will not need to be exported `SourceType` -> `sourceType`